Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -461,32 +382,29 @@ public class HTTPClient {
     ///     - request: HTTP request to execute.
     ///     - delegate: Delegate to process response parts.
     ///     - eventLoop: NIO Event Loop preference.
+    ///     - context: Baggage context associated with this request
     ///     - deadline: Point in time by which the request must complete.
-    ///     - logger: The logger to use for this request.
     public func execute<Delegate: HTTPClientResponseDelegate>(request: Request,
                                                               delegate: Delegate,
                                                               eventLoop eventLoopPreference: EventLoopPreference,
+                                                              context: BaggageContext,
                                                               deadline: NIODeadline? = nil) -> Task<Delegate.Response> {
-        return self.execute(request: request,
-                            delegate: delegate,
-                            eventLoop: eventLoopPreference,
-                            deadline: deadline,
-                            logger: HTTPClient.loggingDisabled)
-    }
+        var span = InstrumentationSystem.tracingInstrument.startSpan(named: request.method.rawValue, context: context, ofKind: .client, at: nil)
+        span.attributes.http.method = request.method.rawValue
+        span.attributes.http.scheme = request.scheme
+        span.attributes.http.target = request.uri
+        span.attributes.http.host = request.host
+        if let requestContentLength = request.body?.length {
+            span.attributes.http.requestContentLength = requestContentLength
+        }
 
-    /// Execute arbitrary HTTP request and handle response processing using provided delegate.
-    ///
-    /// - parameters:
-    ///     - request: HTTP request to execute.
-    ///     - delegate: Delegate to process response parts.
-    ///     - eventLoop: NIO Event Loop preference.
-    ///     - deadline: Point in time by which the request must complete.
-    public func execute<Delegate: HTTPClientResponseDelegate>(request: Request,
-                                                              delegate: Delegate,
-                                                              eventLoop eventLoopPreference: EventLoopPreference,
-                                                              deadline: NIODeadline? = nil,
-                                                              logger originalLogger: Logger?) -> Task<Delegate.Response> {
-        let logger = (originalLogger ?? HTTPClient.loggingDisabled).attachingRequestInformation(request, requestID: globalRequestID.add(1))
+        // TODO: net.peer.ip / Not required, but recommended
+
+        var request = request
+        InstrumentationSystem.instrument.inject(context, into: &request.headers, using: HTTPHeadersInjector())
+
```

Comment:
should be the (new) span context, no?

```swift
InstrumentationSystem.instrument.inject(span.context, into: &request.headers, using: HTTPHeadersInjector())
```

Reply:
@ktoso @slashmo please have a look and let me know if I understand the context propagation in the same way

I think you should inject `span.context` not `context`

See also https://github.com/pokryfka/aws-xray-sdk-swift#context-propagation